torch/_tensor_str.py

```diff
@@ -318,6 +318,12 @@ def _str_intern(inp):
             or (self.device.type == 'cuda' and torch.cuda.current_device() != self.device.index):
         suffixes.append('device=\'' + str(self.device) + '\'')
 
+    # Tensor printing performs tensor operations like slice, indexing, etc to make it in a
+    # representable format. These operations on xla tensor results in recompilations. Hence,
+    # to avoid recompilations, copying the tensor to cpu before printing.
+    if self.device.type == 'xla':
+        self = self.to('cpu')
```

Comment:
@wconstab this looks okay to me, do we just want the check to include lazy as well?
`if self.device.type == 'xla' or self.device.type == 'lazy'`

Reply:
yes, that's what I was thinking.  If you don't want to add it now @aws-rhsoln, I can add it later. Thanks!

Reply:
I can add that check and send in a new commit..